static/js/src/publisher-admin/pages/Releases/AddTrackPanel.tsx

```diff
@@ -0,0 +1,178 @@
+                <>
+                  <Input
+                    help="Version pattern should follow glob pattern"
+                    placeholder="?.v.*"
+                    label="Version pattern"
+                    type="text"
+                    onChange={(e) => setVersionPattern(e.target.value)}
+                    value={versionPattern}
+                  />
+                  <Input
+                    help={
+                      <>
+                        The percentage is a value from 0 to 100
+                        <br />
+                        Releases will be done progressively on the track and its
+                        percentage will be incremented automatically.
+                      </>
+                    }
+                    placeholder="14.9"
+                    label="Automatic phasing percentage"
+                    error={errors?.phasingPercentage}
+                    type="text"
+                    onChange={(e) => {
+                      const phasingPercentageValue = Number(e.target.value);
+                      if (isNaN(phasingPercentageValue)) {
+                        setErrors({
+                          phasingPercentage: "Percentage must be a number",
+                        });
+                      } else if (
+                        phasingPercentageValue &&
+                        (phasingPercentageValue > 100 ||
+                          phasingPercentageValue < 0)
+                      ) {
+                        setErrors({
+                          phasingPercentage:
+                            "Percentage must be between 0 and 100",
+                        });
+                      } else {
+                        setErrors({ ...errors, phasingPercentage: undefined });
+                      }
+                      setPhasingPercentage(e.target.value);
+                    }}
+                    value={phasingPercentage}
+                  />
+                </>
+              ),
+            },
+          ]}
+        />
+        <p>* Mandatory fields</p>
+      </div>
+      <div>
+        <p>
+          <a target="_blank" href="https://juju.is/docs/juju/channel">
+            Learn about tracks
+          </a>{" "}
+          or{" "}
+          <a target="_blank" href="https://discourse.charmhub.io">
+            contact us
+          </a>{" "}
+          for help.
+        </p>
+      </div>
+      <div className="p-panel__footer u-align--right">
+        <div className="u-fixed-width">
+          <Button className="u-no-margin--bottom" onClick={onClose}>
+            Cancel
+          </Button>
+          <Button
+            appearance="positive"
+            onClick={submit}
+            hasIcon={loading}
+            disabled={
+              loading ||
+              !trackName ||
+              !!Object.keys(errors).filter((key) => errors[key]).length
+            }
+          >
+            {loading ? <Spinner text="Add track" /> : "Add Track"}
```

Comment:
When you set loading, can you make the button say "Loading..." instead of keeping it as "Add track" with the loading state? Again, I'm pretty sure we do this on Snapcraft as well so would be good for consistency